client/pages/index.tsx

```diff
@@ -1,10 +1,19 @@
-import styles from "../styles/index.module.css"
-import LandingHeader from "../components/LandingHeader"
+import LandingHeader from "../components/LandingHeader";
+import Auth from "../components/Auth";
+import React, { useState } from "react";
 
-const HomePage = () => (
-    <div className={styles.headerContainer}>
-        <LandingHeader />
-    </div>
-);
+const HomePage = () => {
+    const [display, setDisplay] = useState(true);
+    if (display == true) {
+        return (
+            <LandingHeader
+                onClickHost={() => setDisplay(!display)}
+                onClickJoin={() => console.log("some Join Stuff")}
+            />
+        );
+    } else {
+        return <Auth />;
+    }
```

Comment:
`== true` is redundant for boolean types. Just do `if (bool)` or `if (!bool)`.
Also, nothing wrong with multiple returns, though there's a nice pattern for this with ternary operators to return in a single expression:
```ts
    return display ? (
        <LandingHeader
            onClickHost={() => setDisplay(!display)}
            onClickJoin={() => console.log("some Join Stuff")}
        />
    ) : (
        <Auth />
    );
```